src/base/bittorrent/session.cpp

```diff
@@ -4717,6 +4718,32 @@ void Session::startUpTorrents()
             }
         }
 
+        const TagSet tags = resumeData.tags;
+        for (const QString &tag : tags)
+        {
+            bool isTagRecovered = recoveredTags.contains(tag);
+            if (!tag.isEmpty() && (isTagRecovered || !hasTag(tag)))
```

Comment:
This condition is incorrect. You include recovered tag but ignore it inside body. I think `if (!hasTag(tag))` could be enough.